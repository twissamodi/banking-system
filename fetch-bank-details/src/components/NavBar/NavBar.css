.header-container{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    background-color: #38497E;
}
.header-navbar-list{
    display: flex;
    flex-direction: row;
}
.title{
    color: white;
    padding-left: 20px;
}
.navbar-link{
    color: white;
    text-decoration: none;
    padding-left: 10px;
    padding-right: 10px;
    font-size: larger;
}
.navbar-list-items{
    list-style: none;
    align-items: flex-end;

}
.navbar-icon{
    visibility: hidden;
}
.navbar-content{
    display: flex;
    align-items: flex-end;
    padding-right: 20px;
}
.navbar-links-separator{
    color: white;
}
@media screen and (max-width:600px) {
    .title{
        font-size: 25px;
        padding-left: 0px;
        width: 100%;
        text-align: center;
    }
    .header-container{
        width: 100%;
        text-align: center;
        flex-wrap: wrap;
    }
    .navbar-content{
        padding: 0px;
        width: 100%;
        justify-content: center;
    }
    .header-navbar-list{
        margin-top: 0px;
        padding-left: 0;
    }
}