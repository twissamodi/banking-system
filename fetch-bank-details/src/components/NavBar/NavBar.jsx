import React from 'react';
import './NavBar.css';

function NavBar() {
  return (
    <header className="header-container">
      <h1 className="title">bankBazaar.com</h1>
      <nav className="navbar-content">
        <ul className="header-navbar-list">
          <li className="navbar-list-items">
            <a className="navbar-link" href="/">Home</a>
            <span className="navbar-links-separator">|</span>
          </li>
          <li className="navbar-list-items"><a className="navbar-link" href="/search-by-ifsc">Get bank details</a></li>
        </ul>
      </nav>
    </header>
  );
}

export default NavBar;
