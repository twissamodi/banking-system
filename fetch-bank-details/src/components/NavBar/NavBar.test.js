/* eslint-disable comma-dangle */
/* eslint-disable quotes */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */

import { render, screen } from "@testing-library/react";
import React from "react";
import NavBar from "./NavBar";

describe(NavBar.name, () => {
  test("Should match snapshot", () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });

  test('should display "bankbazaar.com', () => {
    render(<NavBar />);
    screen.getByText("bankBazaar.com");
  });

  test("Click Home goes to Home page", () => {
    render(<NavBar />);
    expect(screen.getByText("Home").closest("a")).toHaveAttribute("href", "/");
  });

  test("Click Get bank details goes to search by IFSC code page", () => {
    render(<NavBar />);
    expect(screen.getByText("Get bank details").closest("a")).toHaveAttribute(
      "href",
      "/search-by-ifsc"
    );
  });
});
