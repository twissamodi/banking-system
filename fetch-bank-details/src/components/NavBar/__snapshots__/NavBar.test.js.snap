// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Should match snapshot 1`] = `
<div>
  <header
    class="header-container"
  >
    <h1
      class="title"
    >
      bankBazaar.com
    </h1>
    <nav
      class="navbar-content"
    >
      <ul
        class="header-navbar-list"
      >
        <li
          class="navbar-list-items"
        >
          <a
            class="navbar-link"
            href="/"
          >
            Home
          </a>
          <span
            class="navbar-links-separator"
          >
            |
          </span>
        </li>
        <li
          class="navbar-list-items"
        >
          <a
            class="navbar-link"
            href="/search-by-ifsc"
          >
            Get bank details
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
